### PR TITLE
Disable OSHMEM layer when no SPMLs available

### PR DESCRIPTION
**Pull request by bwbarrett**
Implement the action item from last devel meeting to not build the OSHMEM layer when there are no SPMLs available (ie, the shmem layer can't possibly work).

### CODE REVIEW

**Comment by bwbarrett** [comment]
Doh!  I forgot about making a tarball in this path.  a good reminder of why we have CI testing :).

File: configure.ac [hunk 1]
```diff
@@ -1106,6 +1106,23 @@ OPAL_MCA
 
 m4_ifdef([project_ompi], [OMPI_REQUIRE_ENDPOINT_TAG_FINI])
 
+# Last minute disable of Open SHMEM if we didn't find any oshmem SPMLs
+if test "$project_oshmem_amc" = "true" -a $OSHMEM_FOUND_WORKING_SPML -eq 0 ; then
```
Comment (jsquyres) [inline configure.ac]: Should this (and the other lines like this in this commit) be `if test ... && test ...`?  I think there's some BSD `test` that doesn't like `-a`/`-o` -- I remember we've gotten in portability kerfuffles with this before.

File: configure.ac [hunk 2]
```diff
@@ -1106,6 +1106,23 @@ OPAL_MCA
...
+    # We don't have an spml that will work, so oshmem wouldn't be able
+    # to run an application.  Therefore, don't build the oshmem layer.
+    if test "$enable_oshmem" != "no" -a -n "$enable_oshmem"; then
+	AC_MSG_WARN([No spml found, so Open SHMEM layer will be non functional.])
```
Comment (jsquyres) [inline configure.ac]: Minor nit: unlike Open MPI, the name is actually "OpenSHMEM" (i.e., no space).

Comment (bwbarrett) [reply configure.ac]: I saw Open SHMEM more than OpenSHMEM in our build code; will fix.

**Comment by jsquyres** [comment]
@bwbarrett It just occurs to me: this PR probably means that we need to install UCX in our CI environments so that OSHMEM gets at least compile tested (e.g., PR #5092 nominally touches the OSHMEM layer, but OSHMEM will no longer compile by default on my platforms because I have no UCX-based networks).

Alternatively, should `--enable-oshmem` enable the OSHMEM layer to be compiled even if there are no viable SPMLs?  This feature might be for development/testing/CI only, perhaps...?

**Comment by bwbarrett** [comment]
I'm pretty sure Mellanox CI tests UCX, so I wasn't worried about that.  I'd rather not add compile-only testing to the EC2 CI and when I tried last, the TCP driver in UCX didn't work.

**Pull request merged**